### ISSUE TITLE
Support IAM instance role

### ISSUE THREAD

**Issue by Mmasson-01** [putyourlightson/craft-blitz-cloudfront#13]
### Bug Report

I've been trying to use the plugin with an IAM instance role (More precisely an AWS IAM ECS instance role) and looking back at the closed issue https://github.com/putyourlightson/craft-blitz-cloudfront/issues/3 it seems i'm either missing something orit was not properly tested.

At the moment, it is not possible to leave the apiKey and apiSecret variable empty as it's `held` by a required rule.

The commit https://github.com/putyourlightson/craft-blitz-cloudfront/commit/170225b5d2009399226ee6a45def4b87eeabdbc8 add the required logic to use an IAM role instead of IAM Secret Keys but forget to remove the **required** rule for the fields.

I believe changing this: 

```php
/**
     * @inheritdoc
     */
    public function rules(): array
    {
        return [
            [['apiKey', 'apiSecret', 'distributionId'], 'required'],
        ];
    }
```

For this: 

```php
/**
     * @inheritdoc
     */
    public function rules(): array
    {
        return [
            [['distributionId'], 'required'],
        ];
    }
```

will do the trick.

I'm open to create a pull request for the changes. I suggest to also update the field label (instricutions) in the twig template to match the official [Pixel & Tonic AWS S3 plugin](https://github.com/craftcms/aws-s3/blob/2.x/README.md) 

![image](https://github.com/user-attachments/assets/f5c625db-ab0a-490a-86e8-04a6636c59d7)

```twig
{{ forms.autosuggestField({
    label: 'API Key'|t('blitz-cloudfront'),
    id: 'apiKey',
    name: 'apiKey',
    instructions: 'You can leave this field empty if you are using an EC2 instance with an applicable IAM role assignment..'|t('blitz-cloudfront'),
    suggestEnvVars: true,
    suggestions: craft.cp.getEnvSuggestions(),
    value: purger.apiKey,
    errors: purger.getErrors('apiKey'),
    required: true,
    first: true,
}) }}

{{ forms.autosuggestField({
    label: 'API Secret'|t('blitz-cloudfront'),
    id: 'apiSecret',
    name: 'apiSecret',
    instructions: 'You can leave this field empty if you are using an EC2 instance with an applicable IAM role assignment.'|t('blitz-cloudfront'),
    suggestEnvVars: true,
    suggestions: craft.cp.getEnvSuggestions(),
    value: purger.apiSecret,
    errors: purger.getErrors('apiSecret'),
    required: true,
}) }}\
```

### Plugin Version

5.9.0

### Craft CMS Version

5.4.6

### PHP Version

8.2

**Comment by bencroker** [comment]
Thanks for reporting this! Fixed in https://github.com/putyourlightson/craft-blitz-cloudfront/commit/8f3e3cd628079b121b796efa905b2d3a50adc21c and released in [5.1.1](https://github.com/putyourlightson/craft-blitz-cloudfront/releases/tag/5.1.1) and [4.1.1](https://github.com/putyourlightson/craft-blitz-cloudfront/releases/tag/4.1.1).

**Comment by Mmasson-01** [comment]
i am flabbergasted by the speed of this fix and release. So glad we're using you plugins at work now ✅🚀 cheers

**Comment by bencroker** [comment]
Well, you spelled it all out for me – much appreciated!

**Comment by Mmasson-01** [comment]
@bencroker while testing the fix on my project at the moment, I forgot the [twig template](https://github.com/putyourlightson/craft-blitz-cloudfront/blob/5.1.1/src/templates/settings.twig) `required: true` function should also be removed to remove the required red asterix on the field label! oops !

![image](https://github.com/user-attachments/assets/0fb92b2e-bf58-447b-a5cd-8b2f4f25bdc7)

Cheers !

**Comment by bencroker** [comment]
Good catch! Fixed in https://github.com/putyourlightson/craft-blitz-cloudfront/commit/d6c974dd40093badedba6aa99d20317087644495 and released in [5.1.2](https://github.com/putyourlightson/craft-blitz-cloudfront/releases/tag/5.1.2) and [4.1.2](https://github.com/putyourlightson/craft-blitz-cloudfront/releases/tag/4.1.2).
